### ISSUE TITLE
Location argument returns wrong location in v1.5

### ISSUE THREAD

**Issue by HielkePlugins** [JorelAli/1.13-Command-API#23]
In version 1.5 the location argument doesn't work properly, It returns the wrong location.

Example:
If I'm at the coordinates 50 10 20 and run:
`/test ~ ~1 ~`
(So the first argument is type location)

The expected location to return is: 50 11 20

But the actual location is I get back is: 50.5 50.5 50.5

**Comment by JorelAli** [comment]
I have been unable to reproduce this issue in version 1.5:

My code:
```java
//Generate arguments and command
LinkedHashMap<String, Argument> arguments = new LinkedHashMap<>();
arguments.put("loc", new LocationArgument());
CommandAPI.getInstance().register("myloc", arguments, (sender, args) -> {
    //Resultant location from argument
    Location loc = (Location) args[0];
    sender.sendMessage("Location argument: (" + loc.getBlockX() + " " + loc.getBlockY() + " " +  loc.getBlockZ() + ")");
	
    //My current location
    Location currentLoc = ((Player) sender).getLocation();
    sender.sendMessage("Current location: (" + currentLoc.getBlockX() + " " + currentLoc.getBlockY() + " " +  currentLoc.getBlockZ() + ")");
    sender.sendMessage(" ");
});
```

Result from the command `/myloc ~ ~1 ~`:

![Screenshot](https://i.imgur.com/RpvPRQm.png "Screenshot")

**Comment by HielkePlugins** [comment]
This is the code I use:
(It's a command that set's the motion of the selector)

    private void motionCommand() {
        LinkedHashMap<String, Argument> arguments = new LinkedHashMap<>();
        arguments.put("motion", new LocationArgument());

        CommandAPI.getInstance().register("motion", arguments, (sender, args) -> {
            CommandSender callee = sender;

            //when using execute, differentiate sender and caller
            if (sender instanceof ProxiedCommandSender) {
                callee = ((ProxiedCommandSender) sender).getCallee();
                sender = ((ProxiedCommandSender) sender).getCaller();
            }

            if (callee instanceof Entity) {
                Entity entity = (Entity) callee;
                Location loc1 = (Location) args[0];
                Location loc2 = entity.getLocation();
                sender.sendMessage("Location argument: (" + loc1.getBlockX() + " " + loc1.getBlockY() + " " +  loc1.getBlockZ() + ")");
                sender.sendMessage("Current location: (" + loc2.getBlockX() + " " + loc2.getBlockY() + " " +  loc2.getBlockZ() + ")");
                Vector motion = loc1.toVector().subtract(loc2.toVector());
                setVelocity(motion, entity, sender);
            }
        });
    }

My output to `/motion ~ ~1 ~` is:
![2018-11-04_01 03 44](https://user-images.githubusercontent.com/42968178/47958557-822bd300-dfcd-11e8-839b-7e6c3ccaba90.png)

In version 1.4 everything works just fine.

**Comment by JorelAli** [comment]
I believe the issue relates to issue #20.
This is resolved in [version 1.6](https://github.com/JorelAli/1.13-Command-API/releases/tag/v1.6).

**Comment by HielkePlugins** [comment]
Yep! Works fine again in 1.6!
